rancher2/structure_cluster_v2_rke_config_certificate_rotation.go

```diff
@@ -0,0 +1,41 @@
+package rancher2
+
+import rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+
+func flattenClusterV2RKEConfigRotateCertificates(in *rkev1.RotateCertificates) []interface{} {
+	if in == nil {
+		return nil
+	}
+
+	obj := make(map[string]interface{})
+
+	if in.Generation > 0 {
+		obj["generation"] = int(in.Generation)
+	}
+
+	if len(in.Services) > 0 {
+		obj["services"] = toArrayInterfaceSorted(in.Services)
+	}
+
+	return []interface{}{obj}
+}
+
+func expandClusterV2RKEConfigRotateCertificates(p []interface{}) *rkev1.RotateCertificates {
+	if p == nil || len(p) == 0 || p[0] == nil {
```

Comment:
Thought for a second here you could condense this, but an array's zero value can be non-nil unlike slices